### DISCUSSION TITLE
Bun v1.1.5

### DISCUSSION THREAD

**Discussion by Jarred-Sumner** [oven-sh/bun#10534] [Announcements]
To install Bun v1.1.5:
```bash
curl -fsSL https://bun.sh/install | bash
# or you can use npm
# npm install -g bun
```

Windows:
```bash
powershell -c "irm bun.sh/install.ps1|iex"
```

To upgrade to Bun v1.1.5:
```bash
bun upgrade
```

## Thank you to 16 contributors!

- [@boyer-victor](https://github.com/boyer-victor)
- [@cirospaciari](https://github.com/cirospaciari)
- [@Deckluhm](https://github.com/Deckluhm)
- [@dylan-conway](https://github.com/dylan-conway)
- [@Electroid](https://github.com/Electroid)
- [@erikbrinkman](https://github.com/erikbrinkman)
- [@gvilums](https://github.com/gvilums)
- [@Jarred-Sumner](https://github.com/Jarred-Sumner)
- [@jwigert](https://github.com/jwigert)
- [@lafkpages](https://github.com/lafkpages)
- [@liudonghua123](https://github.com/liudonghua123)
- [@nektro](https://github.com/nektro)
- [@paperdave](https://github.com/paperdave)
- [@tuttarealstep](https://github.com/tuttarealstep)
- [@welfuture](https://github.com/welfuture)
- [@zackradisic](https://github.com/zackradisic)

**Full Changelog**: https://github.com/oven-sh/bun/compare/bun-v1.1.4...bun-v1.1.5

<hr /><em>This discussion was created from the release <a href='https://github.com/oven-sh/bun/releases/tag/bun-v1.1.5'>Bun v1.1.5</a>.</em>

**Comment by liudonghua123** [comment]
I tried to update bun from 1.1.4 to the latest 1.1.5 using `bun upgrade`, but it failed with the following errors.

![image](https://github.com/oven-sh/bun/assets/2276718/06bd931e-5c57-494d-a267-1654bb1dbf81)

From the error `NOT_SAME_DEVICE` info, maybe it is because the downloaded bun (maybe in temp C:) is not the same disk driver as the original bun (D:\apps\bun.exe).

If I use `npm i -g bun`, it works.

**Reply by liudonghua123** [reply]
I found some code related to this problem.

https://github.com/oven-sh/bun/blob/dfcbe09035e235e0dafe0710d64b83715dae4696/src/cli/upgrade_command.zig#L888-L900

https://github.com/oven-sh/bun/blob/dfcbe09035e235e0dafe0710d64b83715dae4696/src/c.zig#L105-L126

Also maybe related to https://github.com/ziglang/zig/issues/17049.

If the std.os.windows.renameAtW do not support move files across different mounts, can we just replaced to copy-then-remove?

I tested the following code, `std.fs.renameAbsolute` could not move files across different mounts/drivers indeed.

```zig
// src/main.zig
const std = @import("std");

const w = std.os.windows;

pub fn main() !void {
    // move file test across devices
    const src = "C:/Users/Liu.D.H/Desktop/test.txt";
    const dst = "D:/test.txt";
    // try std.fs.renameAbsolute(src, dst); // error.Unexpected: GetLastError(17): The system cannot move the file to a different disk drive.

    // this works!
    try std.fs.copyFileAbsolute(src, dst, .{});
    try std.fs.deleteFileAbsolute(src);
    return;
}

```
